### PR TITLE
Cleanup j2ee.persistence module

### PR DESCRIPTION
**Pull request by pepness**
I did some cleaning to the `j2ee.persistence` module with the `Inspect & Transform` tool. I reviewed each suggestion of the inspection tool with special emphasis on lambdas and try-with-resources and made sure not to break something. After this I plan to improve the JPA support to 2.2 or hopefully 3.1.

- Use lambdas when possible (like new Runnable or calling an async task with RequestProcessor)
- Use try-with-resources
- Use URI.toURL() instead of new URL()
- Use Utilities.toURI method
- Use NbBundle.getMessage instead of NbBundle.getBundle
- Remove redundant if expressions and if statements
- if-else and loops statements should use braces
- Remove unnecessary continue and return statements
- Use Exception constructor when possible
- Use multi-catch
- Unnecessary boxing and primitive instantiation
- Use diamond inference on declaration-initialization
- Use generics
- Use switch expression
- Remove unused logger
- Remove imports from the same package
- Remove unused imports
- Add missing Override annotations
- Fix some typos
- Fix some formatting
- Add a TODO comment

I did full builds and fix the errors that pop out.

NetBeans Testing:
- Full build done (with each commit)
- Verify successful execution of libraries and licenses Ant test
- Verify successful execution of Verify Sigtests
- Verify successful execution of unit tests for module `j2ee.persistence`
- Started NetBeans and ensure the log didn't have any ERROR or new WARNINGS
- Successfully create different persistence units in different application servers.

### CODE REVIEW

**Comment by mbien** [comment]
@pepness I just merged a larger cleanup PR which caused conflicts here. Should be hopefully easy to resolve - bad timing sorry about that.

**Comment by tbw777** [comment]
@pepness 

> Use switch expression

SE was introduced as standartized version only in Java SE 14. Current bytecode version of NB is 8.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/support/JPAAttribute.java [hunk 1]
```diff
@@ -143,55 +143,69 @@ public Class<?> getClass1() {
...
-                    if(declaredType.getTypeArguments()!=null && declaredType.getTypeArguments().size()>0) {//it's some generic type
-                        if(mType == IMappingType.ONE_TO_MANY || mType == IMappingType.MANY_TO_MANY) {//we suppose it should be for relationship mapping only
-                            tm = declaredType.getTypeArguments().get(0);
-                            if(tm.getKind() == TypeKind.DECLARED){
-                                declaredType = (DeclaredType) tm;
+        } else {
+            tm = var.asType();
+        }
+        if( tm != null ){
+            if(null != tm.getKind()) {
```
Comment (matthiasblaesing) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/support/JPAAttribute.java:157]: Would it make sense to pull this up into the upper if? To me these if -> switch changes don't improve readabiliy, but ok.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java [hunk 2]
```diff
@@ -174,21 +174,15 @@ public boolean parseDocument() {
         } else if (isModified()){//if it's isn't modified and persistenc eexits (parsed) no need to reparse
             try{
                 String oldVersion = persistence.getVersion();
-                java.io.InputStream is = getEditorSupport().getInputStream();
                 String version=Persistence.VERSION_1_0;
-                try {
+                try (java.io.InputStream is = getEditorSupport().getInputStream()) {
```
Comment (matthiasblaesing) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java:178]: Could you import `java.io.InputStream`?

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/EntityClosure.java [hunk 3]
```diff
@@ -425,19 +399,9 @@ public boolean isModelReady() {
...
+            Future result = model.runReadActionWhenReady( (EntityMappingsMetadata metadata) -> true );
             result.get();
-        } catch (InterruptedException ex) {
-            Exceptions.printStackTrace(ex);
-        } catch (ExecutionException ex) {
-            Exceptions.printStackTrace(ex);
-        } catch (MetadataModelException ex) {
+        } catch (InterruptedException | ExecutionException | MetadataModelException ex) {
             Exceptions.printStackTrace(ex);
         } catch (IOException ex) {
```
Comment (matthiasblaesing) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/EntityClosure.java:406]: Could be integrated into multi-catch.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/EntityClassesPanel.java [hunk 4]
```diff
@@ -711,7 +703,7 @@ public boolean isValid() {
             }
 
             String packageName = getComponent().getPackageName();
-            if (packageName.trim().equals("")) { // NOI18N
+            if (packageName.trim().isEmpty()) { // NOI18N
```
Comment (matthiasblaesing) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/EntityClassesPanel.java:706]: Drop `NOI18N`

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/SelectedTables.java [hunk 5]
```diff
@@ -196,7 +196,9 @@ public String getClassName(Table table) {
             String exClassName = persistenceGen.getFQClassName(table.getName());
             if(exClassName != null) {
                 int i = exClassName.lastIndexOf('.');
-                if(i>-1)exClassName = exClassName.substring(i+1);
+                if(i>-1) {
```
Comment (matthiasblaesing) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/SelectedTables.java:199]: Formatting

**Comment by matthiasblaesing** [comment]
@pepness thank you. In general looks good, but I left a few nitpicks inline.

**Comment by mbien** [comment]
@pepness wait, where did the commits go. now its only 3. Was that intended?

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/SelectedTables.java (hunk 5 above)
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/SelectedTables.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/EntityClosure.java (hunk 3 above)
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/EntityClosure.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java (hunk 2 above)
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java:178]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/support/JPAAttribute.java (hunk 1 above)
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/support/JPAAttribute.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/provider/ProviderUtil.java [hunk 6]
```diff
@@ -206,14 +207,14 @@ public static DatabaseConnection getConnection(PersistenceUnit pu) {
...
-     * Gets the database connection properties (irl,name,password) specified in the given persistence
+     * Gets the database connection properties (url,name,password) specified in the given persistence
      * unit.
      * 
      * @param pu the persistence unit whose database connection is to 
      * be retrieved; must not be null.
      * 
-     * @return the connection properties specified in the given persistence unit or
-     * <code>null</code> if it didn't specify a connectioh.
+     * @rerturn the connection properties specified in the given persistence unit or
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/provider/ProviderUtil.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/provider/ProviderUtil.java [hunk 7]
```diff
@@ -161,8 +162,8 @@ private static Provider getContainerManagedProvider(Project project) {
      * @param pu the persistence unit whose database connection is to 
      * be retrieved; must not be null.
      * 
-     * @return the connection specified in the given persistence unit or
-     * <code>null</code> if it didn't specify a connectioh.
+     * @rerturn the connection specified in the given persistence unit or
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/provider/ProviderUtil.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/entitygenerator/SQLType.java [hunk 8]
```diff
@@ -145,8 +148,8 @@ private Class getClassForCharType (Integer length, boolean isNullable) {
     // the no-arg getMemberType method are sufficient
     private Class getClassForNumericType(Integer precision, 
             Integer scale, boolean isNullable) {
-        int precValue = ((precision == null) ? -1 : precision.intValue());
-        int scaleValue = ((scale == null) ? -1 : scale.intValue());
+        int precValue = ((precision == null) ? -1 : precision);
+        int scaleValue = ((scale == null) ? -1 : scale);
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/entitygenerator/SQLType.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/db/DBCompletionContextResolver.java [hunk 9]
```diff
@@ -661,9 +685,13 @@ private List completeManyToMany(JPACodeCompletionProvider.Context ctx, CCParser.
                     element:
                     for(VariableElement f : resultFields) {
                         for(javax.lang.model.element.Modifier mod:f.getModifiers()){
-                            if(javax.lang.model.element.Modifier.TRANSIENT.equals(mod))continue element;
+                            if(javax.lang.model.element.Modifier.TRANSIENT.equals(mod)) {
+                                continue element;
+                            }
+                        }
+                        if(JpaControllerUtil.isAnnotatedWith(f,"javax.persistence.Transient")) {
+                            continue;//NOI18N
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/db/DBCompletionContextResolver.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/PUCompletor.java [hunk 10]
```diff
@@ -208,34 +206,24 @@ public List<JPACompletionItem> doCompletion(final CompletionContext context) {
...
+                cc.toPhase(Phase.ELEMENTS_RESOLVED);
+                Project project = FileOwnerQuery.getOwner(fo);
+                EntityClassScopeProvider provider = project.getLookup().lookup(EntityClassScopeProvider.class);
+                EntityClassScope ecs = null;
+                Entity[] entities = null;
+                if (provider != null) {
+                    ecs = provider.findEntityClassScope(fo);
+                }
+                if (ecs != null) {
+                    entities = ecs.getEntityMappingsModel(false).runReadAction( (EntityMappingsMetadata metadata) -> metadata.getRoot().getEntity() );
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/PUCompletor.java:221]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/DocumentContext.java [hunk 11]
```diff
@@ -214,10 +209,7 @@ public boolean equals(Object obj) {
             return false;
         }
         final DocumentContext other = (DocumentContext) obj;
-        if (this.document != other.document && (this.document == null || !this.document.equals(other.document))) {
-            return false;
-        }
-        return true;
+        return !(this.document != other.document && (this.document == null || !this.document.equals(other.document)));
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/DocumentContext.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/ContextUtilities.java [hunk 12]
```diff
@@ -116,18 +116,22 @@ public static String getAttributeTokenImage(DocumentContext context) {
...
      */
     public static String getLocalNameFromTag(String tagName) {
-        if(tagName == null) return null;
-        return (tagName.indexOf(":") == -1) ? tagName : // NOI18N
-            tagName.substring(tagName.indexOf(":")+1, tagName.length()); // NOI18N
+        if(tagName == null) {
+            return null;
+        }
+        return (tagName.indexOf(':') == -1) ? tagName : // NOI18N
+            tagName.substring(tagName.indexOf(':')+1, tagName.length()); // NOI18N
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/ContextUtilities.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/ContextUtilities.java [hunk 13]
```diff
@@ -116,18 +116,22 @@ public static String getAttributeTokenImage(DocumentContext context) {
...
      */
     public static String getPrefixFromTag(String tagName) {
-        if(tagName == null) return null;
-        return (tagName.indexOf(":") == -1) ? null : // NOI18N
-            tagName.substring(0, tagName.indexOf(":")); // NOI18N
+        if(tagName == null) {
+            return null;
+        }
+        return (tagName.indexOf(':') == -1) ? null : // NOI18N
+            tagName.substring(0, tagName.indexOf(':')); // NOI18N
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/ContextUtilities.java]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/PUCompletor.java [hunk 14]
```diff
@@ -208,34 +206,24 @@ public List<JPACompletionItem> doCompletion(final CompletionContext context) {
...
+                }
+                // add classes
+                if(entities != null) {
+                    for (Entity entity : entities) {
+                        if (typedPrefix.length() == 0 || entity.getClass2().toLowerCase().startsWith(typedPrefix.toLowerCase()) || entity.getName().toLowerCase().startsWith(typedPrefix.toLowerCase())) {
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/PUCompletor.java:226]: Done.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/hyperlink/NamedQueryHyperlinkProvider.java [hunk 15]
```diff
@@ -155,73 +152,69 @@ private void goToNQ(Document doc, int offset) {
...
-                                                                foundAm = am;
-                                                                break;
-                                                            }
+                js.runUserActionTask( (CompilationController parameter) -> {
+                    parameter.toPhase(JavaSource.Phase.RESOLVED);
+                    AnnotationMirror foundAm = null;
+                    AnnotationValue get = null;
+                    Trees trees = parameter.getTrees();
+                    
+                    TypeElement entityElement = parameter.getElements().getTypeElement(entClasst);trees.getSourcePositions().getStartPosition(parameter.getCompilationUnit(), trees.getPath(entityElement).getLeaf());
```
[the remark this answers is not in the document]
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/hyperlink/NamedQueryHyperlinkProvider.java:163]: You are correct about the second statement, `trees.getSourcePositions().getStartPosition(parameter.getCompilationUnit(),trees.getPath(entityElement).getLeaf());` 
it is a noop that return a long primitive. Good catch!

**Comment by pepness** [comment]
@mbien I was working on them. Fixed the suggestions that @matthiasblaesing suggest

**Comment by mbien** [comment]
> I was working on them. Fixed the suggestions that @matthiasblaesing suggest

@pepness ah, all good. I was just reviewing it and suddenly most of it was gone :)

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/ETCompletionContextResolver.java [hunk 16]
```diff
@@ -105,12 +104,7 @@ private List<JPACompletionItem> completecreateNamedQueryparameters(JPACodeComple
         }
         if (ecs != null) {
             try {
-                entities = ecs.getEntityMappingsModel(false).runReadAction(new MetadataModelAction<EntityMappingsMetadata, Entity[]>() {
-                   @Override
-                    public Entity[] run(EntityMappingsMetadata metadata) throws Exception {
-                        return metadata.getRoot().getEntity();
-                    }
-                });
+                entities = ecs.getEntityMappingsModel(false).runReadAction( (EntityMappingsMetadata metadata) -> metadata.getRoot().getEntity() );
```
Comment (mbien) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/ETCompletionContextResolver.java:107]: just a nitpick. Consider the one-action-per-line formatting when method chaining is used. Makes the lines shorter.
```java
entities = ecs.getEntityMappingsModel(false)
              .runReadAction( (EntityMappingsMetadata metadata) -> metadata.getRoot().getEntity() );
```
there are many comparable occurrences like this one

**Review by mbien — APPROVED** [review]
looks good to me. Some comments inline for you to consider.

**Comment by mbien** [comment]
just a general comment:
converting big blocks into lambdas basically loses the git blame history of the whole block. So I am not sure how useful that is all things considered. Maybe we shouldn't keep doing this in future. (Maybe consider reverting the largest candidates?)

Converting single char Strings into chars to use a different String method variant is also a stylistic change which has no runtime benefits anymore. Probably something we can skip over in future too.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java [hunk 17]
```diff
@@ -174,21 +175,15 @@ public boolean parseDocument() {
...
                 }
-                finally
-                {
-                    is.close();
-                }
-                is = getEditorSupport().getInputStream();
                 Persistence newPersistence;
                 Persistence cleanPersistence;
-                try {
+                try (java.io.InputStream is = getEditorSupport().getInputStream()) {
```
Comment (matthiasblaesing) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java:186]: This should change the same was a ~20 lines above.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/EntityClosure.java (hunk 3 above)
Comment (matthiasblaesing) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/EntityClosure.java:406]: There is not change here. The IOException still stands alone

**Review by matthiasblaesing — APPROVED** [review]
In general: Lets get this in. Some nitpicks inline, but I don't think this need another round of reviews. Github somehow thought every file was touched between my reviews, which makes it ugly to follow up.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/ETCompletionContextResolver.java (hunk 16 above)
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/ETCompletionContextResolver.java:107]: I like it that way too. Done

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java (hunk 17 above)
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/unit/PUDataObject.java:186]: Done

**Comment by matthiasblaesing** [comment]
Looks sane to me. Thank you.

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/JPAEditorUtil.java [hunk 18]
```diff
@@ -290,7 +288,7 @@ public static DatabaseConnection findDatabaseConnection(PersistenceUnit pu, Proj
             return null;
         }
         List<DatabaseConnection> dbconns = findDatabaseConnections(datasource);
-        if (dbconns.size() > 0) {
+        if (!dbconns.isEmpty()) {
             return dbconns.get(0);
         }
         return null;
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/JPAEditorUtil.java:294]:
```suggestion
        return dbconns.isEmpty()
            ? null
            : dbconns.get(0);
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/db/DBCompletionContextResolver.java [hunk 19]
```diff
@@ -238,7 +246,7 @@ private DatabaseConnection findDatabaseConnection(JPACodeCompletionProvider.Cont
             return null;
         }
         List<DatabaseConnection> dbconns = findDatabaseConnections(datasource);
-        if (dbconns.size() > 0) {
+        if (!dbconns.isEmpty()) {
             return dbconns.get(0);
         }
         return null;
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/db/DBCompletionContextResolver.java:252]:
```suggestion
        return dbconns.isEmpty()
            ? null
            : dbconns.get(0);
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/db/DBMetaDataProvider.java [hunk 20]
```diff
@@ -122,7 +118,9 @@ public synchronized Catalog getCatalog(String name) throws SQLException {
             getCatalogs();
         }
         Catalog ret = (Catalog)catalogs.get(name);
-        if(ret == null && "".equals(name))ret = (Catalog)catalogs.get(null);
+        if(ret == null && "".equals(name)) {
+            ret = (Catalog)catalogs.get(null);
+        }
         return ret;
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/db/DBMetaDataProvider.java:124]:
```suggestion
        return ret == null && "".equals(name))
            ?(Catalog)catalogs.get(null);
            : ret;
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/entitygenerator/DbSchemaEntityMember.java [hunk 21]
```diff
@@ -112,40 +116,50 @@ public void setPrimaryKey(boolean isPk, boolean isPkField) {
...
+                case "double": //NOI18N
+                    return Double.class.getName();
+                case "float": //NOI18N
+                    return Float.class.getName();
+                case "char": //NOI18N
+                    return Character.class.getName();
+                default:
+                    return sqlType.getFirstNonPrimitiveType();
+            }
         }
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/entitygenerator/DbSchemaEntityMember.java:140]:
```suggestion
        switch (type) {
            case "int": //NOI18N
                return Integer.class.getName();
            case "long": //NOI18N
                return Long.class.getName();
            case "short": //NOI18N
                return Short.class.getName();
            case "byte": //NOI18N
                return Byte.class.getName();
            case "double": //NOI18N
                return Double.class.getName();
            case "float": //NOI18N
                return Float.class.getName();
            case "char": //NOI18N
                return Character.class.getName();
            default:
                return sqlType.getFirstNonPrimitiveType();
        }
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/provider/ProviderUtil.java [hunk 22]
```diff
@@ -595,22 +596,31 @@ public static Provider getProvider(PersistenceUnit persistenceUnit, Provider[] p
     private static Set<Provider> filterProvidersByProperties(Set<Provider> providers, Property[] properties){
         Set<Provider> ret = null;
         if(providers == null){}
-        else if(providers.size()<= 1 || properties==null || properties.length==0) ret = new HashSet(providers);
-        else {
+        else if(providers.size()<= 1 || properties==null || properties.length==0) {
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/provider/ProviderUtil.java:599]:
```suggestion
        else if(providers.size() <= 1 || properties == null || properties.length == 0) {
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/provider/ProviderUtil.java [hunk 23]
```diff
@@ -595,22 +596,31 @@ public static Provider getProvider(PersistenceUnit persistenceUnit, Provider[] p
...
+                        rate++;
+                    }
+                }
+                if(rates.get(rate) == null) {
+                    rates.put(rate, new ArrayList<Provider>());
                 }
-                if(rates.get(rate) == null)rates.put(rate, new ArrayList<Provider>());
                 rates.get(rate).add(each);
-                if(rate<lowrate)lowrate=rate;
+                if(rate<lowrate) {
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/provider/ProviderUtil.java:616]:
```suggestion
                if(rate < lowrate) {
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/Constructor.java [hunk 24]
```diff
@@ -91,7 +91,7 @@ private TypeDeclaration typeToTypeDeclaration(TypeMirror tMirror){
         if(aType.getKind() == TypeKind.DECLARED){
             DeclaredType dType = (DeclaredType) aType;
             List<? extends TypeMirror> parameters = dType.getTypeArguments();
-            if( parameters!=null && parameters.size()>0){
+            if( parameters!=null && !parameters.isEmpty()){
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/Constructor.java:94]:
```suggestion
            if( parameters != null && !parameters.isEmpty()){
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/support/JPAAttribute.java [hunk 25]
```diff
@@ -166,32 +167,43 @@ private void buildType(){
                     }
                     typeElement =  (TypeElement) declaredType.asElement();
                     typeName = typeElement.getQualifiedName().toString();
-                } else if (TypeKind.BOOLEAN == tm.getKind()) {
+                    break;
+                case BOOLEAN:
                     typeName = ("boolean");//NOI18N
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/spi/jpql/support/JPAAttribute.java:172]:
```suggestion
                    typeName = "boolean";//NOI18N
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/PersistenceClientEntitySelectionVisual.java [hunk 26]
```diff
@@ -600,7 +584,9 @@ public Component getListCellRendererComponent(JList list, Object value, int inde
                 setBackground(list.getBackground());
                 setForeground(list.getForeground());
             }
-            if(text.length() == 0)text = " ";
+            if(text.length() == 0) {
+                text = " ";
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/PersistenceClientEntitySelectionVisual.java:588]:
```suggestion
                text = ' ';
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/DBSchemaManager.java [hunk 27]
```diff
@@ -151,32 +150,39 @@ public void run(final ProgressSupport.Context actionContext) {
...
-                            message = NbBundle.getMessage(DBSchemaManager.class, "LBL_RetrievingViewKeys", event.getNewValue());
-                        }
-                        
-                        if (message != null) {
-                            actionContext.progress(message);
+                schemaElementImpl.addPropertyChangeListener( (PropertyChangeEvent event) -> {
+                    String propertyName = event.getPropertyName();
+                    String message = null;
+                    
+                    if (null != propertyName) {
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/DBSchemaManager.java:157]:
```suggestion
                    if (propertyName != null) {
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/EntityClassesPanel.java [hunk 28]
```diff
@@ -373,9 +360,14 @@ private void updateSetAllButtons(){
         {
             for (Table table : selectedTables.getTables()) {
                 if(!selectedTables.getUpdateType(table).equals(UpdateType.NEW)){
-                    if(selectedTables.getUpdateType(table).equals(UpdateType.UPDATE))recreate=true;
-                    else update=true;
-                    if(update && recreate)break;
+                    if(selectedTables.getUpdateType(table).equals(UpdateType.UPDATE)) {
+                        recreate=true;
+                    } else {
+                        update=true;
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/fromdb/EntityClassesPanel.java:366]:
```suggestion
                        recreate = true;
                    } else {
                        update = true;
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/jpacontroller/JpaControllerGenerator.java [hunk 29]
```diff
@@ -226,488 +219,501 @@ private static FileObject generateJpaController(
...
+                allRelMethods.addAll(toManyRelMethods);
+                
+                String[] importFqs = {"javax.persistence.Query",
+                    "javax.persistence.EntityNotFoundException"
+                };
+                
+                for (String importFq : importFqs) {
+                    modifiedImportCut = JpaControllerUtil.TreeMakerUtils.createImport(workingCopy, modifiedImportCut, importFq);
+                }
+                if(version!=null && !Persistence.VERSION_1_0.equals(version)){//add criteria classes if appropriate
```
Comment (noelma) [inline java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/jpacontroller/JpaControllerGenerator.java:340]:
```suggestion
                if(version != null && !Persistence.VERSION_1_0.equals(version)){//add criteria classes if appropriate
```

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/ETCompletionContextResolver.java (hunk 16 above)
Comment (mbien) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/editor/completion/ETCompletionContextResolver.java]: @pepness there are some more invocations of this method in the same style which you converted to lambdas btw. You changed only the one occurrence so far if I see this correctly.

search for `getEntityMappingsModel(false).` in
https://patch-diff.githubusercontent.com/raw/apache/netbeans/pull/5444.diff

File: java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/jpacontroller/JpaControllerGenerator.java (hunk 29 above)
Comment (pepness) [reply java/j2ee.persistence/src/org/netbeans/modules/j2ee/persistence/wizard/jpacontroller/JpaControllerGenerator.java:340]: Thanks for the suggestions, I will create another PR with better formatting, meanwhile lets get this merged.

**Pull request merged**